Dockerfile

```diff
@@ -5,38 +5,31 @@ WORKDIR /app
 ARG UID=901
 
 RUN wget -q https://raw.githubusercontent.com/vishnubob/wait-for-it/master/wait-for-it.sh -P /usr/local/bin \
-&& chmod +x /usr/local/bin/wait-for-it.sh \
-&& mkdir -p /app /var/lib/document-merge-service/data /var/lib/document-merge-service/media /var/www/static \
-&& useradd -u $UID -r document-merge-service --create-home \
-&& mkdir /home/document-merge-service/.config \
-&& chmod -R 770 /var/lib/document-merge-service/data /var/lib/document-merge-service/media /var/www/static /home/document-merge-service \
-&& apt-get update && apt-get install -y --no-install-recommends unoconv libreoffice-writer && rm -rf /var/lib/apt/lists/* \
-# All project specific folders need to be accessible by newly created user but also for unknown users (when UID is set manually). Such users are in group root.
-&& chown -R document-merge-service:root /var/lib/document-merge-service/data /var/lib/document-merge-service/media /var/www/static /home/document-merge-service
+  && chmod +x /usr/local/bin/wait-for-it.sh \
+  && mkdir -p /app /var/lib/document-merge-service/data /var/lib/document-merge-service/media /var/www/static \
+  && useradd -u $UID -r document-merge-service --create-home \
+  && mkdir /home/document-merge-service/.config \
+  && chmod -R 770 /var/lib/document-merge-service/data /var/lib/document-merge-service/media /var/www/static /home/document-merge-service \
+  && apt-get update && apt-get install -y --no-install-recommends util-linux unoconv libreoffice-writer && rm -rf /var/lib/apt/lists/* \
+  # All project specific folders need to be accessible by newly created user but also for unknown users (when UID is set manually). Such users are in group root.
+  && chown -R document-merge-service:root /var/lib/document-merge-service/data /var/lib/document-merge-service/media /var/www/static /home/document-merge-service
+RUN pip install poetry
 
 # Needs to be set for users with manually set UID
 ENV HOME=/home/document-merge-service
+USER document-merge-service
 
 ENV PYTHONUNBUFFERED=1
 ENV DJANGO_SETTINGS_MODULE document_merge_service.settings
 ENV APP_HOME=/app
 ENV UWSGI_INI /app/uwsgi.ini
-ENV STATIC_ROOT /var/www/static
 ENV MEDIA_ROOT /var/lib/document-merge-service/media
 
 ARG ENV=docker
 COPY pyproject.toml poetry.lock $APP_HOME/
-RUN pip install poetry
-RUN poetry config virtualenvs.create false \
-  && poetry install $([ "$ENV" = "dev" ] || echo "--no-dev") --no-interaction --no-ansi
+RUN poetry install $([ "$ENV" = "dev" ] || echo "--no-dev") --no-interaction --no-ansi
 COPY . $APP_HOME
 
-RUN ENV=docker ./manage.py collectstatic --noinput \
-&& chown -R document-merge-service:root /var/www/static \
-&& chmod 770 -R /var/www/static \
-&& cp /bin/sleep /bin/dms_test_sleep
-
-USER document-merge-service
 EXPOSE 8000
 
-CMD /bin/sh -c "./manage.py migrate && uwsgi"
```

Comment:
Does this work? Should be `poetry run uwsgi`, no?

Reply:
No, I switched to virtualenvs with poetry again. If you install it into system, everything is fine, ,except you also install dev-dependencies, then it goes and remove everything from system, according to the poetry.lock. Including itself.

This was in another project, I don't actually know  why `poetry config virtualenvs.create false` works here. But I decided, we should be consistent and not add `poetry config virtualenvs.create false`.

Reply:
I think the whole line should be this:
`CMD /bin/sh -c "./manage.py migrate && poetry run uwsgi"` as uwsgi is installed via Poetry and wouldn't be found without it.. You might need to test it without the `docker-compose.override.yml` file as it overrides the cmd portion with the dev server (which is indeed run with poetry)